### PR TITLE
libbsd: fixup on darwin

### PR DESCRIPTION
**Pull request by matthewbauer**
This is my attempt to get libbsd workin on Darwin. It's kind of a mess
right now with patches. Most of it are ugly hacks but at least it builds!

Should fix issue #23434
###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/libbsd/default.nix [hunk 1]
```diff
@@ -1,18 +1,33 @@
...
+
+  preAutoreconf = "mkdir m4";
+
+  patchPhase = ''
+    substituteInPlace configure.ac \
+      --replace "m4_esyscmd([./get-version])" "${version}"
+    sed -i '38i#undef strlcpy' include/bsd/string.h
+    sed -i '38i#undef strlcat' include/bsd/string.h
+    substituteInPlace src/setproctitle.c \
+     --replace 'extern typeof(setproctitle_impl) setproctitle_stub __attribute__((weak, alias("setproctitle_impl")));' ""
```
Comment (peterhoeg) [inline pkgs/development/libraries/libbsd/default.nix]: Please turn this into a patch instead.

Comment (matthewbauer) [inline pkgs/development/libraries/libbsd/default.nix]: ok I just combined everything into one patch. That way we won't change the hash for Linux systems.

**Comment by peterhoeg** [comment]
Much nicer, thanks! 👍

**Comment by LnL7** [comment]
@peterhoeg Do you remember what we changed to work around this?

**Review by LnL7 — APPROVED** [review]

**Comment by peterhoeg** [comment]
@LnL7, we didn't. We simple avoided the use of ```libbsd```.

**Pull request merged**